### PR TITLE
Cleanup expired forward compatibility horizons

### PR DESCRIPTION
**Pull request by lgeiger**
The forward compatibility horizons are expired since last year, so these checks can be safely removed.

### CODE REVIEW

**Comment by bhack** [comment]
It could be nice if CI could check this automatically and a Github bot open a ticket. 
/cc @tensorflow/api-owners

I have this candidate list for 2020
```
tensorflow/python/compat/compat_test.py:45:      self.assertTrue(compat.forward_compatible(2020, 4, 4))
tensorflow/python/debug/lib/debug_v2_ops_test.py:241:    if not compat.forward_compatible(2020, 6, 24):
tensorflow/python/debug/lib/dumping_callback.py:381:      if tf_compat.forward_compatible(2020, 6, 24):
tensorflow/python/debug/lib/dumping_callback.py:384:      if tf_compat.forward_compatible(2020, 7, 1):
tensorflow/python/keras/layers/preprocessing/image_preprocessing.py:680:    if compat.forward_compatible(2020, 8, 5):
tensorflow/python/ops/stateful_random_ops.py:544:    if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateful_random_ops.py:585:    if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateful_random_ops.py:669:    if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateful_random_ops.py:715:    if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateful_random_ops.py:727:    if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateful_random_ops.py:799:        if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/sparse_ops.py:645:  if separator is None and not tf_compat.forward_compatible(2020, 6, 14):
tensorflow/python/ops/image_ops_impl.py:3126:    if compat.forward_compatible(2020, 8, 14):
tensorflow/python/ops/image_ops_impl.py:5288:      tile_size == 512 and not compat.forward_compatible(2020, 6, 23):
tensorflow/python/ops/embedding_ops.py:560:      if compat.forward_compatible(2020, 5, 14):
tensorflow/python/ops/stateless_random_ops.py:216:      if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateless_random_ops.py:229:        if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateless_random_ops.py:238:        if compat.forward_compatible(2020, 10, 25):
tensorflow/python/ops/stateless_random_ops.py:568:    if compat.forward_compatible(2020, 10, 25):
tensorflow/python/distribute/input_lib_test.py:858:    if not reshuffle and not compat.forward_compatible(2020, 5, 22):
tensorflow/python/data/experimental/ops/readers.py:792:    if compat.forward_compatible(2020, 7, 3) or exclude_cols is not None:
tensorflow/python/data/experimental/ops/error_ops.py:66:    if compat.forward_compatible(2020, 8, 26) or log_warning:
tensorflow/compiler/tests/stateful_random_ops_test.py:162:    if compat.forward_compatible(2020, 10, 25):
tensorflow/compiler/tests/stateful_random_ops_test.py:181:    if compat.forward_compatible(2020, 10, 25):
tensorflow/compiler/tests/stateful_random_ops_test.py:218:    if compat.forward_compatible(2020, 10, 25):
tensorflow/compiler/tests/stateful_random_ops_test.py:261:    if compat.forward_compatible(2020, 10, 25):
```

And 2021 candidates
```
tensorflow/python/ops/stateless_random_ops.py:127:  if compat.forward_compatible(2021, 3, 1):
tensorflow/python/ops/stateless_random_ops.py:518:    if compat.forward_compatible(2021, 3, 1):
tensorflow/python/ops/lookup_ops.py:753:      if self._offset != 0 or compat.forward_compatible(2021, 3, 18):
```

**Review by jsimsa — APPROVED** [review]

File: tensorflow/python/ops/image_ops_impl.py [hunk 1]
```diff
@@ -3192,22 +3191,15 @@ def decode_image(contents,
...
-            dtype=dtype)
+    channels = 0 if channels is None else channels
+    if dtype not in [dtypes.float32, dtypes.uint8, dtypes.uint16]:
+      dest_dtype = dtype
+      dtype = dtypes.uint16
+      return convert_image_dtype(gen_image_ops.decode_image(
+          contents=contents,
+          channels=channels,
+          expand_animations=expand_animations,
+          dtype=dtype), dest_dtype)
```
Comment (hyeygit) [inline tensorflow/python/ops/image_ops_impl.py:3203]: You would need to remove lines 3204-3276 and remove line 3195, and keep the rest (with proper indentation). The else block removed above is not the else case for `if compat.forward_compatible(2020, 8, 14)`.

**Review by hyeygit — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/image_ops_impl.py (hunk 1 above)
Comment (lgeiger) [reply tensorflow/python/ops/image_ops_impl.py]: Good catch, sorry about that. Looks like my editor got confused with the 2-space indent.

Comment (hyeygit) [reply tensorflow/python/ops/image_ops_impl.py]: Thank you for the quick fix!

**Review by hyeygit — APPROVED** [review]

**Pull request merged**